File: abc/abc159/abc159_d.cs
// https://atcoder.jp/contests/abc159/submissions/11109035

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var N = long.Parse(Console.ReadLine());
            var A = Console.ReadLine().Split().Select(long.Parse).ToArray();

            var B = new long[N + 1];
            for (var i = 0; i < N; i++)
            {
                B[A[i]]++;
            }

            var sum = 0L;
            for (var i = 1; i < N + 1; i++)
            {
                sum += B[i] * (B[i] - 1) / 2;
            }

            for (int i = 0; i < N; i++)
            {
                Console.WriteLine(sum - (B[A[i]] - 1));
            }
        }
    }
}

File: abc/abc159/abc159_d.fs
// https://atcoder.jp/contests/abc159/submissions/11755736

let N = stdin.ReadLine() |> int
let A = stdin.ReadLine().Split() |> Array.map int

let B = Array.zeroCreate<int64> (N + 1)

A
|> Array.groupBy (fun n -> n)
|> Array.map (fun (k, v) ->
    B.[k] <- (v
              |> Array.length
              |> int64))
|> ignore

let sum =
    B
    |> Array.map (fun x -> x * (x - 1L) / 2L)
    |> Array.sum

[| 0..(N - 1) |]
|> Array.map (fun i -> sum - (B.[A.[i]] - 1L))
|> Array.iter (printfn "%d")
